packages/host/tests/helpers/index.gts

```diff
@@ -670,7 +683,66 @@ export function diff(
   };
 }
 
-export class MockResponse extends Response {
+function isCardSourceFetch(request: Request) {
+  return (
+    request.method === 'GET' &&
+    request.headers.get('Accept') === SupportedMimeType.CardSource &&
+    request.url.includes(testRealmURL)
+  );
+}
+
+export async function sourceFetchReturnUrlHandle(
+  request: Request,
+  defaultHandle: (req: Request) => Promise<Response | null>,
+) {
+  if (isCardSourceFetch(request)) {
+    let r = await defaultHandle(request);
+    if (r) {
+      return new MockRedirectedResponse(r.body, r, request.url) as Response;
+    }
+  }
+  return null;
+}
+
+export async function sourceFetchRedirectHandle(
+  request: Request,
+  adapter: RealmAdapter,
+) {
+  let urlParts = request.url.split('.');
```

Comment:
Won’t this split on the “.” In the origin? Like cardstack.com? Perhaps you should turn the url into a `URL` and split on the path like:
`new URL(request.url).path.split(“.”)`